### PR TITLE
Improve ECA documentation in CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by 0xdaryl**
Make ECA requirements more prominent in the contribution guidelines.
Includes minor edits to miscellaneous text for readability.

This was feedback from a recent OMR release review.

### CODE REVIEW

**Comment by 0xdaryl** [comment]
@mstoodle : would you mind reviewing please?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -331,26 +337,18 @@ If the hook declines your commit, the message will remain in
...
-1. [Create an account at Eclipse](https://dev.eclipse.org/site_login/createaccount.php).
-2. Add your GitHub user name in your account settings.
-3. [Log into the project's portal](https://projects.eclipse.org/) and sign the ["Eclipse ECA"](https://projects.eclipse.org/user/sign/cla).
-4. Ensure that you author the commit using the _same_ email address as your Eclipse account.
-5. Include the appropriate copyright notice and license at the top of each file.
+When a pull request is created (or when new commits are pushed to an existing pull
+request), a webservice will automatically run that will verify that each commit
+author's email address matches an email address that signed the ECA.
+**This service is case-sensitive, so the case of the email addresses must match exactly.**
+The success or failure of this webservice will be reported in the GitHub pull request.
```
Comment (mstoodle) [inline CONTRIBUTING.md:348]: The only thing I would add here is a more explicit comment that pull requests that fail the ECA check will not be merged.

**Review by mstoodle — CHANGES_REQUESTED** [review]
Mostly looks very good, thanks for pulling this together

File: CONTRIBUTING.md (hunk 1 above)
Comment (0xdaryl) [reply CONTRIBUTING.md:348]: Fixed in forced push.

**Review by mstoodle — APPROVED** [review]

**Pull request merged**